Framework/API/inc/MantidAPI/Algorithm.h

```diff
@@ -306,6 +307,34 @@ class MANTID_API_DLL Algorithm : public IAlgorithm,
   /// Virtual method - must be overridden by concrete algorithm
   virtual void exec() = 0;
 
+  virtual void execDistributed() { exec(); }
+  virtual void execMasterOnly();
+  virtual void execNonMaster(){};
```

Comment:
Non-master refers to all ranks *but* the master rank. This function is executed on non-master ranks, when to master calls `exec()`. I needed this in 1 or 2 special cases, where despite having no data (since all data is on the master rank), the other ranks still have to do something (like initialize and output workspace).